src/aoc2020/day3_copy.clj

```diff
@@ -0,0 +1,112 @@
+(ns aoc2020.day3-copy
+  (:require [aoc2020.util :refer [file->seq]]))
+
+(defn tree-at-pos?
+  [grid position]
+  (let [row      (first position)
+        col      (second position)
+        location (-> row
+                     grid
+                     (get col))]
+    (= \# location)))
+
+(defn position-s
+  [grid slope-dist]
+  (let [row-number (count grid)
+        col-number (count (first grid))
+        row-s      (->> (iterate (fn [row]
+                                   (+ row (slope-dist :down)))
+                                 0)
+                        (take-while (fn [row] (< row row-number))))
+        col-s      (->> (iterate (fn [col]
+                                   (-> col
+                                       (+ (slope-dist :right))
+                                       (mod col-number)))
+                                 0)
+                        (take (count row-s)))]
+    (map (fn [row col] [row col])
+         row-s
+         col-s)))
+
+(defn tree-count
+  [grid slope-dist]
+  (let [position-s (position-s grid slope-dist)]
+    (->> (filter (partial tree-at-pos? grid)
+                 position-s)
+         (count))))
+
+
+
+(defn product-of-tree-count
+  [grid slope-dist-s]
+  (->> (map (fn [slope-dist]
+              (tree-count grid slope-dist))
+            slope-dist-s)
+       (reduce *)))
+
+
+(comment
+  (do (def sample-grid (file->seq "aoc2020/day3/sample-input.txt")) sample-grid)
+  #_=> ["..##......."
+        "#...#...#.."
+        ".#....#..#."
+        "..#.#...#.#"
+        ".#...##..#."
+        "..#.##....."
+        ".#.#.#....#"
+        ".#........#"
+        "#.##...#..."
+        "#...##....#"
+        ".#..#...#.#"]
+
+  (do (def grid (file->seq "aoc2020/day3/input.txt")) grid)
+  #_=> ["...........#..#.#.###....#....."
+        "...#..#...........#.#...#......"
+        "#.....#..#........#...#..##...."
+        "..#...##.#.....#.............#."
+        "#.#..#......#.....#....#......."
+        ".....#......#..#....#.....#...."
+        ".......#.#..............#......"
+        ".....#...#..........##...#.....",,,]
+
+  ;; part 1
+
+  (def slope-dist {:right 3 :down 1})
+
+  ;; generate the position[row col] sequence
+  (position-s sample-grid slope-dist)
```

Comment:
Why would a function that generates a list of positions require a grid if it's not answer any question about the grid?